pyramid_oereb/lib/records/image.py

```diff
@@ -20,4 +33,113 @@ def encode(self):
         Returns:
             str: The encoded image.
         """
-        return base64.b64encode(self.content).decode('ascii')
+        return b64.encode(self.content)
+
+    @staticmethod
+    def _validate_filetype(obj):
+        """
+        Checks for valid file type and returns extension and mime type.
+
+        Args:
+            obj (str, bytes, bytearray or readable): Path to the file, a readable object or the file's content
+                as str, bytes or bytearray. See filetype documentation:
+                https://h2non.github.io/filetype.py/v1.0.0/filetype.m.html
+
+        Returns:
+            tuple: The file's extension and mime type.
+
+        Raises:
+            TypeError: Raised if type is invalid or could not be recognized.
+        """
+
+        try:
+            ft = filetype.guess(obj)
+        except IOError:
+            ft = None
+
+        # Check for SVG content if filetype.guess fails
```

Comment:
A question regarding backwards compatibility:
could this change behaviour, if png is not detected for some reason in an installation (whereas the file is really png)?
If so, we should add a note of caution in the migration notes.

Reply:
Yes, you are right. It is a kind of restriction that affects backwards compatibility, if someone uses other image formats than PNG or SVG. We should add something in the migration notes.

Reply:
I have added some migration notes with a warning concerning the image formats:
https://github.com/openoereb/pyramid_oereb/pull/917/commits/9b1c910d671fbc1fb1a9d40dc3155b8f82c63f4f#diff-f86adfaa9b2056d4a7b2d6624899961dR10-R23

Reply:
Thanks